quinn/src/connection.rs

```diff
@@ -751,8 +704,6 @@ impl ConnectionRef {
             closed: Arc::new(Notify::new()),
             error: None,
             ref_count: 0,
-            udp_state,
```

Comment:
I've split out the `udp_state` refactor; good call there. Staring at this for a bit I'm not sure there's any other low hanging fruit to decompose. The heart of this commit is folding the responsibilities of `ConnectionDriver` into `EndpointDriver`. This requires restructuring how connections are woken (via the new channel) and switching from timer-per-connection to a `DelayQueue` (since we no longer have a discrete task to wake for each connection). Because timeouts can wake connections, and woken connections can adjust timers, I'm not sure these can be easily decoupled.

Reply:
So the new channel is what you've called `dirty`, right? That's kind of where I struggle, making sense of what's going dirty and how the mechanics for that could make sense. Do you think the documentation/naming could be improved to clarify how this stuff fits together? Like, `dirty` is a nice short name but it doesn't say much about what that state applies to and/or when it's cleared/how it should it should be cleared.

Reply:
> Do you think the documentation/naming could be improved to clarify how this stuff fits together?

I've added additional comments on this subject in several places, including `EndpointInner::dirty_recv`, `ConnectionInner::dirty`, and `Endpoint::drive_connections` (formerly `handle_events`). I do think "dirty" is well precedented for describing the pattern of tracking a subset of components that need to be updated, but better documentation never hurts.

> Sorry to be a drag on this review

To the contrary, your feedback has been plenty timely, especially on the time scale that I've been chipping away at this! Maybe you didn't notice but addressing your last round of comments was spread over a week or so :D

If anything, a reduced-context perspective is helpful, since we do want the architecture to be intelligible to others.